lighthouse-core/gather/driver.js

```diff
@@ -832,9 +857,11 @@ class Driver {
 
     // Wait for load or timeout and run the cleanup function the winner returns.
```

Comment:
edit this comment to note that `securityCheckPromise` will only ever reject?

Reply:
OR crazy idea, the promise resolves with a `cleanupFn` that throws an error :D then the three cases are a little more parallel

Reply:
to your second comment, so `securityCheckPromise` would be changed to `resolve` instead of `reject`?

Reply:
> to your second comment, so securityCheckPromise would be changed to resolve instead of reject?

right it would go hand-in-hand with https://github.com/GoogleChrome/lighthouse/pull/6608#discussion_r246261703 to enable to refactor